### ISSUE TITLE
[Ascend950PR][SIMD] TTIR-to-Linalg converter segfaults on scalar tt.extern_elementwise isinf (f32 -> i1)

### ISSUE THREAD

**Issue by maxxlife** [triton-lang/triton-ascend#1401]
### Describe the bug

**Environment:**
```
torch: 2.7.1+cpu
torch_npu: 2.7.1.post5+git091ff3c
triton: 3.2.0
```
**Description:**
The Triton-Ascend compiler crashes with a seg fault compiling on a scalar **tl.extra-cann-libdevice.isin**f operation using `compile_mode="simd"` on Ascend 950PR.
The crash occurs during compile-only warmup(), before the kernel is launched. The same kernel compiles successfully with `compile_mode="simt_only"`.

**Minimal reproducer version 1:**
`repro_isinf_950_v1.py`
```
import os
import sys

mode = sys.argv[1] if len(sys.argv) > 1 else "simt_only"
assert mode in {"simd", "simt_only"}

# Must be selected before importing Triton.
if mode == "simt_only":
    os.environ["TRITON_ENABLE_LIBDEVICE_SIMT"] = "1"
else:
    os.environ.pop("TRITON_ENABLE_LIBDEVICE_SIMT", None)

os.environ["TRITON_ALWAYS_COMPILE"] = "1"
os.environ["TRITON_KERNEL_DUMP"] = "1"
os.environ["TRITON_DUMP_DIR"] = f"/tmp/triton-isinf-{mode}"

import torch
import torch_npu
import triton
import triton.language as tl


@triton.jit
def scalar_isinf_kernel(x_ptr, out_ptr):
    # Each Triton program processes one scalar, matching issue #213583.
    pid = tl.program_id(0)
    x = tl.load(x_ptr + pid)
    result = tl.extra.cann.libdevice.isinf(x)
    tl.store(out_ptr + pid, result.to(tl.int32))


finfo = torch.finfo(torch.float32)
x_cpu = torch.tensor(
    [
        float("-inf"),
        float("inf"),
        float("nan"),
        -finfo.max,
        -1.0,
        -0.0,
        0.0,
        1.0,
        finfo.max,
    ],
    dtype=torch.float32,
)

expected = torch.isinf(x_cpu).to(torch.int32)
x = x_cpu.npu()
actual_npu = torch.full(x.shape, -1, dtype=torch.int32, device="npu")

print("mode:", mode)
print("torch:", torch.__version__)
print("torch_npu:", getattr(torch_npu, "__version__", "unknown"))
print("triton:", triton.__version__)

scalar_isinf_kernel[(x.numel(), 1, 1)](
    x,
    actual_npu,
    compile_mode=mode,
)
torch.npu.synchronize()

actual = actual_npu.cpu()
print("expected:", expected.tolist())
print("actual:  ", actual.tolist())

assert torch.equal(actual, expected)
print("RESULT: PASS")
```

**Minimal reproducer version 2:**
`repro_isinf_950_v2.py`
```
import os
import sys

mode = sys.argv[1] if len(sys.argv) > 1 else "simt_only"
assert mode in {"simd", "simt_only"}

# Must be selected before importing Triton.
if mode == "simt_only":
    os.environ["TRITON_ENABLE_LIBDEVICE_SIMT"] = "1"
else:
    os.environ.pop("TRITON_ENABLE_LIBDEVICE_SIMT", None)

os.environ["TRITON_ALWAYS_COMPILE"] = "1"
os.environ["TRITON_KERNEL_DUMP"] = "1"
os.environ["TRITON_DUMP_DIR"] = f"/tmp/triton-isinf-{mode}"

import torch
import torch_npu
import triton
import triton.language as tl


@triton.jit
def scalar_isinf_kernel(x_ptr, out_ptr):
    # Each Triton program processes one scalar, matching issue #213583.
    pid = tl.program_id(0)
    x = tl.load(x_ptr + pid)
    result = tl.extra.cann.libdevice.isinf(x)
    tl.store(out_ptr + pid, result.to(tl.int32))


finfo = torch.finfo(torch.float32)
x_cpu = torch.tensor(
    [
        float("-inf"),
        float("inf"),
        float("nan"),
        -finfo.max,
        -1.0,
        -0.0,
        0.0,
        1.0,
        finfo.max,
    ],
    dtype=torch.float32,
)

expected = torch.isinf(x_cpu).to(torch.int32)
x = x_cpu.npu()
actual_npu = torch.full(x.shape, -1, dtype=torch.int32, device="npu")

print("mode:", mode)
print("torch:", torch.__version__)
print("torch_npu:", getattr(torch_npu, "__version__", "unknown"))
print("triton:", triton.__version__)

# Compile the kernel without launching it.
scalar_isinf_kernel.warmup(
    x,
    actual_npu,
    grid=(x.numel(), 1, 1),
    compile_mode=mode,
)

print("COMPILE-ONLY: PASS")
```

**Execution Results**
**SIMD Mode:**
Command: `python repro_isinf_950_v1.py simd` python repro_isinf_950_v2.py simd`
Result: Segmentation fault (core dumped)
Status: Failure (Crashes before TTAdapter/Linalg MLIR generation)

**SIMT-Only Mode:**
Command: `python repro_isinf_950_v1.py simt_only` `python repro_isinf_950_v2.py simt_only`
Result: COMPILE-ONLY: PASS
Status: Success (Correctly produces TTIR and npubin)

**Observed Behavior**
`simd mode:` Crashes with Segmentation fault (core dumped). The process terminates after TTIR generation but before producing TTAdapter/Linalg MLIR.
`simt_only mode:` Compiles successfully, generating both TTIR and npubin.

**Technical Analysis**
The failed SIMD compilation produces TTIR but crashes before producing TTAdapter/Linalg MLIR.
The successful SIMT-only compilation produces both TTIR and npubin.
The SIMD and SIMT-only TTIR inputs are identical. The relevant operation is:
```
The TTIR for the failing operation is:
%3 = tt.extern_elementwise %2 {
  libname = "",
  libpath = "",
  pure = true,
  symbol = "__hmf_isinf_fp32"
} : (f32) -> i1
```
The target is:
`#hacc.target<"Ascend950PR_9579">`

This appears to localize the failure to the Triton-Ascend SIMD TTIR-to-Linalg/TritonAdapter conversion, before the AscendNPU compiler. In particular, the converter may not correctly handle scalar tt.extern_elementwise with the signature (f32) -> i1.

**Expected Behavior**
The compiler should either successfully generate the kernel for the SIMD path or provide a clear, descriptive diagnostic error if the operation is unsupported, rather than terminating via segmentation fault.

### Environment details

Hardware: Ascend 950PR
Triton: 3.2.0
Torch: 2.7.1+cpu
Torch_NPU: 2.7.1.post5+git091ff3c

**Comment by hongziqi** [comment]
Thanks for your feedback. We will arrange relevant engineers to investigate this issue later.

**Comment by zhuxuejie8** [comment]
I have tested these two cases using TA version 3.2.2 and TA version 3.6, and both passed in simd and simt mode。
You can directly install the TA 3.2.2 by running the following command: pip install triton-ascend --extra-index-url=https://mirrors.huaweicloud.com/ascend/repos/pypi.

**Comment by hongziqi** [comment]
We will close this issue if there are no further concerns. Feel free to open a new issue if you encounter new problems later.
